### PR TITLE
angular: gave a more precise return type for IQService.all()

### PR DESCRIPTION
**Pull request by kalleguld**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [X] Increase the version number in the header if appropriate. (No version information present)
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. (minor edit)

### CODE REVIEW

**Comment by sandersn** [comment]
Mapped types requires typescript 2.3, so the error log shows lots of places that need be explicit about their 2.3 dependency.

File: types/angular/index.d.ts [hunk 1]
```diff
@@ -1029,8 +1029,7 @@ declare namespace angular {
          *
          * @param promises A hash of promises.
          */
-        all(promises: { [id: string]: IPromise<any>; }): IPromise<{ [id: string]: any; }>;
-        all<T extends {}>(promises: { [id: string]: IPromise<any>; }): IPromise<T>;
+        all<T extends {}>(promises: { [K in keyof T]: IPromise<T[K]>; }): IPromise<T>;
```
Comment (sandersn) [inline types/angular/index.d.ts]: Can you add or modify tests to show how this change alters what can be done with the new signature for `all`? 

I found that the old version detects objects with non-Promise-wrapped properties, but of course returns a much worse type. The new version returns a nice type, but doesn't catch incorrect objects with non-Promise-wrapped properties. You just get a bogus type: `IPromise<{}>`. I want to make sure that you're aware of the tradeoff and are OK with it.

Comment (kalleguld) [inline types/angular/index.d.ts]: Good points, thanks for having a look. I have added a few tests. 

I have also fixed the signature so the function also takes non-promise properties.

File: types/angular-block-ui/angular-block-ui-tests.ts [hunk 2]
```diff
@@ -1,3 +1,6 @@
+// TypeScript Version: 2.3
```
Comment (sandersn) [inline types/angular-block-ui/angular-block-ui-tests.ts]: from looking at the Travis error log, it seems that the version annotation goes only in index.d.ts

**Pull request merged**